pkg/crd/crd.go

```diff
@@ -315,46 +315,54 @@ func (c *Client) syncCache(exit <-chan struct{}, initial bool, cacheSyncs ...cac
 }
 
 // RemoveAssignedIdentity removes the assigned identity
-func (c *Client) RemoveAssignedIdentity(assignedIdentity *aadpodid.AzureAssignedIdentity) error {
+func (c *Client) RemoveAssignedIdentity(assignedIdentity *aadpodid.AzureAssignedIdentity) (err error) {
 	glog.V(6).Infof("Deletion of assigned id named: %s", assignedIdentity.Name)
 	begin := time.Now()
-	err := c.rest.Delete().Namespace(assignedIdentity.Namespace).Resource("azureassignedidentities").Name(assignedIdentity.Name).Do().Error()
-	glog.V(5).Infof("Deletion %s took: %v", assignedIdentity.Name, time.Since(begin))
-	stats.Update(stats.AssignedIDDel, time.Since(begin))
 
-	c.reporter.Report(
-		metrics.AssignedIdentityDeletionCountM.M(1),
-		metrics.AssignedIdentityDeletionDurationM.M(metrics.SinceInSeconds(begin)))
+	defer func() {
+		if err != nil {
+			c.reporter.ReportKubernetesAPIOperationError(metrics.AssignedIdentityDeletionOperationName)
+			return
+		}
+		c.reporter.Report(
+			metrics.AssignedIdentityDeletionCountM.M(1),
```

Comment:
Here we are differencing in the pattern - we are counting the deletes instead of looking at the number from the latency counters. What do you say ?